### PR TITLE
Pre coach-contents work

### PR DESCRIPTION
**Pull request by jonboiser**
Pre coach-contents cleanup:

1. Optimize and add comments to ContentNodeGranularSerializer @lyw07 
1. Add a `coach-content-label` component that will be used in multiple areas to indicate a thing is "Coach Content"
1. Reskin tables in import/export to use CoreTable
1. Remove "mappers" that were basically identity function from Learn app
1. Mechanical refactorings (use named imports/exports, one import per line in Python, remove temp variables, etc.)
1. Break up vuex stuff in Learn page into separate modules; remove test utilities that were needed because vuex stuff was not broken up.
1. Update buble and buble-loader. Remove `prettier-ignore` comments that were meant to prevent bad buble code transformation.
1. added `id` to ContentNode serializers to aid in using `id` consistently in UI code #3072 
1. use less expensive `.exists()` based query for `ChannelMetadata.has_exercise filter` @alanchenz . and add tests for the filter

### CODE REVIEW

File: kolibri/content/serializers.py [hunk 1]
```diff
@@ -304,66 +327,82 @@ def to_representation(self, instance, progress_fraction=None, annotate_progress_
...
-        model = ContentNode
-        fields = (
-            'pk', 'content_id', 'title', 'description', 'kind', 'available', 'sort_order', 'license_owner',
-            'license_name', 'license_description', 'files', 'parent', 'author',
-            'assessmentmetadata', 'lang', 'channel_id',
-        )
-
-        list_serializer_class = ContentNodeListSerializer
+    def get_has_coach_content(self, instance):
+        return False
```
Comment (lyw07) [inline kolibri/content/serializers.py]: Sorry I'm not following here. Why does `get_has_coach_content` always return `False`? Thanks!

Comment (jonboiser) [reply kolibri/content/serializers.py]: that was a "stub", but I've removed it after seeing how it was implemented in STudio

Comment (lyw07) [reply kolibri/content/serializers.py]: thank you for the clarification!

**Comment by jonboiser** [comment]
@lyw07 Thanks for taking a look at the python changes. I'll get someone else to look at the JS.

File: kolibri/plugins/device_management/assets/src/views/select-content-page/content-node-row.vue [hunk 2]
```diff
@@ -87,11 +89,6 @@
 
 <style lang="stylus" scoped>
 
-  .row
-    display: table
-    width: 100%
-    vertical-align: middle
-
   .display-cell
     display: table-cell
```
Comment (christianmemije) [inline kolibri/plugins/device_management/assets/src/views/select-content-page/content-node-row.vue:93]: Not necessary. Same goes for the width style below.

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 3]
```diff
@@ -344,35 +265,28 @@ export function showTopicsContent(store, id) {
...
+      });
       store.dispatch('CORE_SET_PAGE_LOADING', false);
     })
     .catch(error => {
       handleApiError(store, error);
     });
 }
 
-export function clearSearch(store) {
+export function clearSearch(store, searchTerm = '') {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js:285]: When would searchTerm not be `''`?

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 4]
```diff
@@ -344,35 +265,28 @@ export function showTopicsContent(store, id) {
...
-  searchResultsPromise
+  return ContentNodeResource.getPagedCollection({ search: searchTerm })
+    .fetch()
     .then(results => {
-      const searchState = { searchTerm };
-      searchState.contents = _collectionState(results);
-      store.dispatch('SET_PAGE_STATE', searchState);
+      store.dispatch('SET_PAGE_STATE', {
+        searchTerm,
+        content: _collectionState(results),
```
Comment (jonboiser) [inline kolibri/plugins/learn/assets/src/state/actions/main.js:276]: this should be `contents`, will be fixed in follow-up

**Pull request merged**